packages/core/src/rules/oas3/component-name-unique.ts

```diff
@@ -46,26 +49,27 @@ export const ComponentNameUnique: Oas3Rule | Oas2Rule = (options) => {
           const resolvedRef = resolve(ref);
           if (!resolvedRef.location) return;
 
-          addComponentFromAbsoluteLocation(
-            typeName,
-            resolvedRef.location.absolutePointer.toString()
-          );
+          addComponentFromAbsoluteLocation(typeName, resolvedRef.location);
         }
       },
     },
     Root: {
       leave(root: Oas3Definition, ctx: UserContext) {
         components.forEach((value, key, _) => {
-          if (value.size > 1) {
+          if (value.absolutePointers.size > 1) {
             const component = getComponentFromKey(key);
             const optionComponentName = getOptionComponentNameForTypeName(component.typeName);
-            const definitions = Array.from(value)
+            const definitions = Array.from(value.absolutePointers)
               .map((v) => `- ${v}`)
               .join('\n');
-
             const problem: Problem = {
               message: `Component '${optionComponentName}/${component.componentName}' is not unique. It is defined at:\n${definitions}`,
             };
+            problem.location = {
+              ...value.locations.sort((a, b) =>
+                a.absolutePointer.localeCompare(b.absolutePointer)
```

Comment:
What's the reason of using `localeCompare`?

Reply:
I wanted to keep the code simple. I am not aware of a `compare` function that is without the `locale` part. Even though the locale part is not needed. But a simple `===` won't work as we need -1, 0, 1 return values for it to sort corectly.